src/setuptools_pyproject_migration/__init__.py

```diff
@@ -54,6 +55,12 @@ def initialize_options(self):
     def finalize_options(self):
         pass
 
+    @staticmethod
+    def _transform_contributors(name_string: Optional[str], email_string: Optional[str]) -> List[Contributor]:
+        names = map(str.strip, (name_string or "").split(","))
+        emails = map(str.strip, (email_string or "").split(","))
+        return [{"name": n, "email": e} for n, e in itertools.zip_longest(names, emails, fillvalue="") if n or e]
```

Comment:
So the `if n or e` is where we come stuck in Python ≤ 3.9.

A quick way I can think of fixing this (may not be the most elegant):

`any([s and (s != "UNKNOWN") for s in (n, e)])`

This seems to have the logic we want:

```
RC=0 stuartl@rikishi /tmp/mockpkg $ ipython3
Python 3.11.4 (main, Jul 14 2023, 09:30:02) [GCC 12.2.1 20230428]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.12.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: defined = lambda n, e : any([s and (s != "UNKNOWN") for s in (n, e)])

In [2]: defined("foo", "bar@example.com")
Out[2]: True

In [3]: defined("foo", None)
Out[3]: True

In [4]: defined(None, "bar@example.com")
Out[4]: True

In [5]: defined(None, None)
Out[5]: False

In [6]: defined("UNKNOWN", None)
Out[6]: False

In [7]: defined(None, "UNKNOWN")
Out[7]: False

In [8]: defined("UNKNOWN", "UNKNOWN")
Out[8]: False
```

Reply:
Oh that's sneaky :laughing: Thanks for chasing this down!

The thing that comes to mind with that approach is that it will still emit the record if the name is present but the email address isn't, or vice versa, and then we could get something like `{"name": "foo", "email": "UNKNOWN"}`. And I don't think it makes sense to use `"UNKNOWN"` as a value in any context unless we're consistently using it everywhere, which we're not. (Maybe we should? :shrug: I think not, unless we find a good reason to.) With that in mind I would rather capture the `"UNKNOWN"`s in the name string and email string and replace them with `None`. Let me implement that, I'll push another commit and let you know.

Reply:
Pushed f80e21cb6a6618153f0af38d0721a21c38bccd78 with a fix, which I will need to squash before merging, but feel free to take a look and let me know what you think.